File: compiler/bench/over4.c
// expected: 200000000
// Over on depth 4, 200M times
#include <stdio.h>

int main(void) {
    long a = 0, b = 1, c = 2, d = 3;
    for (long i = 0; i < 200000000L; i++) {
        long over = a;  // over from depth 4
        a = over + 1;
    }
    (void)b;
    (void)c;
    (void)d;
    printf("%ld\n", a);
    return 0;
}

File: compiler/bench/over4.fs
\ expected: 200000000
\ Over on depth 4, 200M times
: main
  0 1 2 3                   ( a b c d )
  200000000 0 do
    >r >r over r> r>        ( a b a c d ) over from depth 4
    drop nip swap >r swap r> ( a' b c d )
    >r >r >r 1 + r> r> r>
  loop
  drop drop drop . cr
;
